tangelo/problem_decomposition/dmet/dmet_problem_decomposition.py

```diff
@@ -545,30 +574,115 @@ def _compute_energy(self, mf_frag, onerdm, twordm, fock_frag_copy, t_list, onein
         norb = t_list[0]
 
         # Calculate the one- and two- RDM for DMET energy calculation (Transform to AO basis)
-        one_rdm = reduce(np.dot, (mf_frag.mo_coeff, onerdm, mf_frag.mo_coeff.T))
+        one_rdm = mf_frag.mo_coeff @ onerdm @ mf_frag.mo_coeff.T
 
         twordm = np.einsum("pi,ijkl->pjkl", mf_frag.mo_coeff, twordm)
         twordm = np.einsum("qj,pjkl->pqkl", mf_frag.mo_coeff, twordm)
         twordm = np.einsum("rk,pqkl->pqrl", mf_frag.mo_coeff, twordm)
         twordm = np.einsum("sl,pqrl->pqrs", mf_frag.mo_coeff, twordm)
 
         # Calculate the total energy based on RDMs
-        total_energy_rdm = np.einsum("ij,ij->", fock_frag_copy, one_rdm) + 0.5 * np.einsum("ijkl,ijkl->", twoint,
-                                                                                           twordm)
+        total_energy_rdm = np.einsum("ij,ij->", fock_frag_copy, one_rdm)
+        + 0.5 * np.einsum("ijkl,ijkl->", twoint, twordm)
```

Comment:
I'd be shocked if that worked. You had to end your line with `\` for all the others, and I just don't understand how Python would parse a new line that just says

`+ 0.5 * np.einsum("ijkl,ijkl->", twoint, twordm)`

Reply:
It is fixed now. However, it did not matter because `total_energy_rdm` is not used to computed the total energy afterwards (the value is ignored, but it could be checked for debugging purposes).